pallet-chainlink-feed/src/lib.rs

```diff
@@ -730,15 +762,20 @@ pub mod pallet {
 
 				// update oracle rewards and try to reserve them
 				let payment = details.payment;
-				T::Currency::reserve(&T::PalletId::get().into_account(), payment).or_else(
+				// track the debt in case we cannot reserve
+				T::Currency::reserve(&Self::account_id(), payment).or_else(
 					|_| -> DispatchResult {
 						// track the debt in case we cannot reserve
-						Debt::<T>::try_mutate(|debt| {
-							*debt = debt.checked_add(&payment).ok_or(Error::<T>::Overflow)?;
+						Self::try_mutate_debt(feed_id, |debt, max_debt| {
+							let new_debt =
+								debt.checked_add(&payment).ok_or(Error::<T>::Overflow)?;
+							ensure!(new_debt < max_debt, <Error<T>>::MaxDebtReached);
+							*debt = new_debt;
```

Comment:
This operation doesn't work in the test `can_go_into_debt_and_repay`, have checked

* Remove the wrapper `with_transaction_result` ( The result of the wrapper is Ok(..) in the test )
* Composed a new extrinsic only contains these lines ( It works )
* Force write these lines into the test directly ( it works )
* Force write these lines as the last lines of this call ( doesn't work )

Please help to review this

Reply:
@clearloop the `with_transaction_result` is currently necessary because we have fallible functions that modify storage.
i was not able to detect any obvious error/problem after looking at this briefly.
However, I wonder if we actually need these `mutate_debt` functions, we're tracking the `debt` within the `FeedConfig` and we already read it from storage. 
Instead, wouldn't it be sufficient to apply the addition directly to the feed we have already read and then write it back to storage?
This way would get rid of those functions entirely.

Reply:
https://github.com/ChainSafe/chainlink-polkadot/blob/ffcdd64610a5d85e66c2c86b9cb87a89bd695ff4/pallet-chainlink-feed/src/lib.rs#L1519

This problem is caused by this fucking `Drop` implementation